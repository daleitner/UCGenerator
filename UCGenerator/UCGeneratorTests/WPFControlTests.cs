using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using UCGenerator;
using UCGenerator.Statics;

namespace UCGeneratorTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class WPFControlTests
	{
		[TestMethod]
		public void WhenCreateWPFControl_VerifyBindings()
		{
			var control = new WPFControl();
			Approvals.VerifyAll(control.Bindings.Select(x => x.PropertyName), "Binding");
		}

		[TestMethod]
		public void WhenChangeType_UpdateBindings()
		{
			var control = new WPFControl();
			control.Type = TypeEnum.Button;
			Approvals.VerifyAll(control.Bindings.Select(x => x.PropertyName), "Binding");
		}
	}
}
