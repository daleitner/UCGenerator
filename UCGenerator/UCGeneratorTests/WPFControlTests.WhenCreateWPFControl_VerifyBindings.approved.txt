Binding[0] = Text
Binding[1] = Property-Changed-Trigger
Binding[2] = IsEnabled
Binding[3] = Visibility
Binding[4] = Key-Binding
